### PR TITLE
Add FOAMMOB and friends

### PR DESCRIPTION
**Pull request by atgeirr**
Also includes FOAMOPTS and WFOAM. Supporting code for WFOAM including test will be in a separate pull request.

### CODE REVIEW

**Comment by atgeirr** [comment]
jenkins build this please

File: src/opm/parser/eclipse/share/keywords/000_Eclipse100/W/WFOAM [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "name" : "WFOAM",
+  "sections" : [ "SPECIAL", "SCHEDULE" ],
```
Comment (joakim-hove) [inline src/opm/parser/eclipse/share/keywords/000_Eclipse100/W/WFOAM]: The `SPECIAL` section is a random 3+ years old error which is copy & pasted from PR to PR. Just remove it.

File: src/opm/parser/eclipse/share/keywords/000_Eclipse100/W/WFOAM [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+  "items" :
+    [{"name" : "WELL"                        , "value_type" : "STRING"},
+     {"name" : "FOAM_CONCENTRATION"          , "value_type" : "UDA" , "dimension" : "FoamDensity"}
```
Comment (joakim-hove) [inline src/opm/parser/eclipse/share/keywords/000_Eclipse100/W/WFOAM]: Lo and behold - you are forward leaning; you have a plan here?

**Comment by joakim-hove** [comment]
:heavy_check_mark:

File: src/opm/parser/eclipse/share/keywords/000_Eclipse100/W/WFOAM (hunk 1 above)
Comment (atgeirr) [reply src/opm/parser/eclipse/share/keywords/000_Eclipse100/W/WFOAM]: Will do.

File: src/opm/parser/eclipse/share/keywords/000_Eclipse100/W/WFOAM (hunk 2 above)
Comment (atgeirr) [reply src/opm/parser/eclipse/share/keywords/000_Eclipse100/W/WFOAM]: Not really, beyond trying to not use yesterday's tech here? To me it seems supporting UDA here is no more effort than straight doubles.

Comment (joakim-hove) [reply src/opm/parser/eclipse/share/keywords/000_Eclipse100/W/WFOAM]: I guess you are right - so the simple solution here will just be to have `"UDA"` in the json file - as you have, and then when you internalize it in some `EclipseState` object you do:
```
double foam_concentration = record.getItem("FOAM_CONCENTRATION").get<UDAValue>(0).get<double>();
```
which will throw if the input is not a plain double, and then at a next stage we need to pass context to said object to *really* enable UDQ.

**Comment by atgeirr** [comment]
Updated to remove the extra SPECIAL sections.

**Comment by atgeirr** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
Just merge when jenkins has done it's extra round.

File: src/opm/parser/eclipse/share/keywords/000_Eclipse100/W/WFOAM (hunk 2 above)
Comment (atgeirr) [reply src/opm/parser/eclipse/share/keywords/000_Eclipse100/W/WFOAM]: I see. That means UDQ will not really be enabled for this yet, but the groundwork has been done.

Comment (joakim-hove) [reply src/opm/parser/eclipse/share/keywords/000_Eclipse100/W/WFOAM]: That is correct - the parser will handle both `"STRING"` and `0.26` and the parsed data will end up in a `UDAValue` instance, but when fetching data from that instance you must have a plan (i.e. a context available) if you are to handle string/UDA data.

**Pull request merged**